amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RecordsPublisher.java

```diff
@@ -40,15 +49,41 @@
      * @param recordsRetrieved the processRecordsInput to restart from
      */
     void restartFrom(RecordsRetrieved recordsRetrieved);
-    
+
 
     /**
      * Shutdowns the publisher. Once this method returns the publisher should no longer provide any records.
      */
     void shutdown();
 
+    /**
+     * Gets last successful response details.
+     *
+     * @return details associated with last successful response.
+     */
+    Optional<RequestDetails> getLastSuccessfulResponseDetails();
+
+    /**
+     * Gets last successful response's request id.
+     *
+     * @return requestId associated with last succesful response.
+     */
+    default String getLastSuccessfulResponseRequestId() {
+        return getLastSuccessfulResponseDetails().map(RequestDetails::requestId).orElse(NONE);
+    }
+
+    /**
+     * Gets last successful response's timestamp.
+     *
+     * @return timestamp associated with last successful response.
+     */
+    default String getLastSuccessfulResponseTimestamp() {
+        return getLastSuccessfulResponseDetails().map(RequestDetails::timestamp).orElse(NONE);
```

Comment:
Let's move this to RequestDetails if these methods are required. Ideally we should be requiring only toString() of RequestDetails.

Reply:
Yeah I think that makes more sense. Updated.